File: dotnet/rapyd-sample/models/PaymentMethodRequiredFields.cs
using System.Text.Json.Serialization;

namespace rapyd_sample.models
{
    public class Field
    {
        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("type")]
        public string Type { get; set; }

        [JsonPropertyName("regex")]
        public string Regex { get; set; }

        [JsonPropertyName("is_required")]
        public bool IsRequired { get; set; }

        [JsonPropertyName("instructions")]
        public string Instructions { get; set; }
    }

    public class PaymentMethodOption
    {
        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("type")]
        public string Type { get; set; }

        [JsonPropertyName("regex")]
        public string Regex { get; set; }

        [JsonPropertyName("description")]
        public string Description { get; set; }

        [JsonPropertyName("is_required")]
        public bool IsRequired { get; set; }

        [JsonPropertyName("is_updatable")]
        public bool IsUpdatable { get; set; }
    }

    public class PaymentMethodRequiredFields
    {
        [JsonPropertyName("type")]
        public string Type { get; set; }

        [JsonPropertyName("fields")]
        public List<Field> Fields { get; set; }

        [JsonPropertyName("payment_method_options")]
        public List<PaymentMethodOption> PaymentMethodOptions { get; set; }

        [JsonPropertyName("payment_options")]
        public List<PaymentOption> PaymentOptions { get; set; }

        [JsonPropertyName("minimum_expiration_seconds")]
        public object MinimumExpirationSeconds { get; set; }

        [JsonPropertyName("maximum_expiration_seconds")]
        public object MaximumExpirationSeconds { get; set; }
    }
}

File: nodejs/src/models/paymentMethodRequiredFields.ts
import { Type } from 'class-transformer';
import { IsArray, IsBoolean, IsInt, IsString, ValidateNested } from 'class-validator';

export class Field {
  @IsString()
  name: string;
  @IsString()
  type: string;
  @IsString()
  regex: string;
  @IsBoolean()
  is_required: boolean;
  @IsString()
  instructions: string;
}

export class PaymentMethodOption {
  @IsString()
  name: string;
  @IsString()
  type: string;
  @IsString()
  regex: string;
  @IsString()
  description: string;
  @IsBoolean()
  is_required: boolean;
  @IsBoolean()
  is_updatable: boolean;
}

export class PaymentOption {
  @IsString()
  name: string;
  @IsString()
  type: string;
  @IsString()
  regex: string;
  @IsString()
  description: string;
  @IsBoolean()
  is_required: boolean;
  @IsBoolean()
  is_updatable: boolean;
}

export class PaymentMethodRequiredFields {
  @IsString()
  type: string;
  @IsArray({ each: true })
  @ValidateNested({ each: true })
  @Type(() => Field)
  fields: Field[];
  @IsArray({ each: true })
  @ValidateNested({ each: true })
  @Type(() => PaymentMethodOption)
  payment_method_options: PaymentMethodOption[];
  @IsArray({ each: true })
  @ValidateNested({ each: true })
  @Type(() => PaymentOption)
  payment_options: PaymentOption[];
  @IsInt()
  minimum_expiration_seconds: number;
  @IsInt()
  maximum_expiration_seconds: number;
}
